features/step_definitions/table_steps.rb

```diff
@@ -17,16 +17,22 @@
     r.replace([*row, r.join(' ')])
   end
 
-  begin
-    expected.diff! table
-  rescue Cucumber::MultilineArgument::DataTable::Different, IndexError => error
-    if ENV['CI']
-      puts error.message
-      puts expected.to_s
-    end
-
-    raise
+  retries ||= 1
+
+  expected.diff! table
+rescue Cucumber::MultilineArgument::DataTable::Different, IndexError => error
+  if retries > 0
+    retries -= 1
+    sleep 1
+    retry
+  end
```

Comment:
This retrying is to prevent clicking something and then browser being just a little too slow to update before we perform the check. Not super useful in all scenarios but it should not slow down test suite in general, only prevent some false negatives.

In particular I saw [features/api/alerts.feature:Reading alerts](https://github.com/akostadinov/porta/blob/43424c3d42e4b1913efec5e1eb84fc27e346966e/features/api/alerts.feature#L39) feature fail.

Attaching error logs and png. You can see that cucumber detected the first raw not marked as read but in the PNG that is collected at teardown, it was already marked as read. So I expect this repetition to prevent such click+check table errors.

[logs.zip](https://github.com/3scale/porta/files/14027399/logs.zip)

Reply:
A bit more context: this is caused because there is an async action performed by javascript (table action "Read" in this case). Capybara is designed to wait for JS requests but checks the table in a way that we skip this waiting time.

Other solutions that are more explicit:
- add step `wait a moment` after clicking an async JS action.
- add waiting time before calling `extract_table` to give it time to update (maybe more time if env is Oracle)
- (in the future) refactor this step so that it takes advantage of capybara's `default_max_wait_time`.